### ISSUE TITLE
Basic definition of ResourceManager API

### ISSUE THREAD

**Issue by hendrikebbers** [hashgraph/hedera-services#7413]
This issue describes the draft of the minimal & initial `ResourceManager` interface. The interface should be used as base for all concurrent tasks in platform and services. In general the manager provides `Executor` implementations that can be used to execute tasks in parallel. Since an `Executor` can be implemented in several different ways (single threaded, thread pool, virtual threads) the config api will be used internally. Therefore each `Executor` is defined by a name. Since we want some additional features for our executors we extend the `Executor` interface with our custom `ExtendedExecutor` interface.

## ResourceManager

The ResourceManager interface is defined like this:

```
public interface ResourceManager {

    @NonNull
    ExtendedExecutor getOrCreateExecutor(@NonNull String uniqueName);

}
```

Internally the implementation will check if an executor with the given `uniqueName` already exists. In that case the existing executor instance will be returned. Otherwise a new instance will be created. The `uniqueName` that is passed to the create method is used internally to create a configured executor. The implementation will use the config api internally. Let's assume an executor is called `foo`, in that case the executor can be accessed like this:

```
ExtendedExecutor fooExecutor = resourceManager. getOrCreateExecutor("foo");
```

Internally that code will use the config API to get the configuration for the executor. Today the concrete & final config parameters do not need to be defined. That will be done in another issue. In general you can assume that the following configuration parameters might be used to configure the executor:

```
resourceManager.executors.foo.name
resourceManager.executors.foo.threadCount
```

Once the concrete parameters are defined the implementation will contain a config record that holds all parameters.

## ExtendedExecutor

The `java.util.concurrent.Executor` interface is very limited and therefore people often use the `java.util.concurrent.ExecutorService` interface when working with concurrency. Sadly that interface contains some methods that we do not won't to provide for our executors: `void shutdown()` allows to shutdown an executor. Since all our executors are managed by the `ResourceManager` and can be used by multiple places we do not want to just call `shutdown()` on them.

> **Warning**
> We need to define how executors can be stopped. Several of the executors that we will need run endless but some might be short running and we need to have a good way how to stop them..

The `ExtendedExecutor` interface looks like that:

```
public interface ExtendedExecutor extends Executor {

    String getName();

    boolean isSingleThreaded();

    boolean isVirtualThreaded();

    Future<Void> submit(Runnable task);

    <V> Future<V> submit(Callable<V> task);
}
```

**Comment by poulok** [comment]
Platform implemented a new framework that may make these resource manager tickets moot. If we need something like this in the future, we will create new tickets.
